pkg/policy/l4.go

```diff
@@ -783,32 +789,43 @@ const (
 	OriginatingTLS TLSDirection = "originating"
 )
 
+// getCerts reads certificates out of the PolicyContext, reading from k8s or local files depending on config
+// and puts the values into the relevant keys in the TLSContext. Note that if the returned TLSContext.FromFile is
+// `false`, then this will be read from Kubernetes.
 func (l4 *L4Filter) getCerts(policyCtx PolicyContext, tls *api.TLSContext, direction TLSDirection) (*TLSContext, error) {
 	if tls == nil {
 		return nil, nil
 	}
-	ca, public, private, err := policyCtx.GetTLSContext(tls)
+	ca, public, private, inlineSecrets, err := policyCtx.GetTLSContext(tls)
```

Comment:
It sort of seems like the interface of `GetTLSContext` doesn't to change for what you want to achieve, and it almost looks like a lot of extra code and interface updates that could be skipped. Instead, could you check if ` tlsCtx.Secret` is `nil` here instead?